### PR TITLE
Docs - Update SuperBench documents

### PR DESCRIPTION
**Pull request by abuccts**
Update SuperBench documents.

__Major Revisions__
* add introduction
* add getting started
* add developer guides
* add CLI api

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,304 +9,14 @@
...
-
-If you would like to contribute to the project, please follow below steps of joint development on GitHub.
-
-1. `Fork` the repo first to your personal GitHub account.
-2. Checkout from main branch for feature development.
-3. When you finish the feature, please fetch the latest code from origin repo, merge to your branch and resolve conflict.
-4. Submit pull requests to origin main branch.
-5. Please note that there might be comments or questions from reviewers. It will need your help to update the pull request.
+* Introduction
+* Getting Started
```
Comment (TobeyQin) [inline README.md:19]: Can we add links on "Introduction" and "Getting Started"? Then user can access the content easily.

File: docs/cli.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+              | |
+              |_|
+
+Welcome to the SB CLI!
+```
+
+## SuperBench CLI commands
+
+Below is the full list of SuperBench CLI commands and their usages:
+
```
Comment (TobeyQin) [inline docs/cli.md:26]: Do we need to add "sb help"?

File: README.md (hunk 1 above)
Comment (TobeyQin) [reply README.md:19]: Do we need to add "Developer Guides"?

File: docs/cli.md (hunk 2 above)
Comment (abuccts) [reply docs/cli.md:26]: updated

File: README.md (hunk 1 above)
Comment (abuccts) [reply README.md:19]: we don't have a link yet
no need to add entry for every category?

**Review by TobeyQin — APPROVED** [review]

File: docs/introduction.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  * Provide comprehensive performance comparison between different existing hardware
+  * Provide insights for hardware and software co-design
+
+It includes micro-benchmark for primitive computation and communication benchmarking,
+and model-benchmark to measure domain-aware end-to-end deep learning workloads.
+
+:::note
+SuperBench is in the early pre-alpha stage for open source, and not ready for general public yet.
+If you want to jump in early, you can try building latest code yourself.
+:::
```
Comment (cp5555) [inline docs/introduction.md:27]: MD cannot know this semantics?

File: docs/getting-started/run-superbench.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+---
+id: run-superbench
+---
+
+# Run SuperBench
+
+Having prepared benchmark configuration and inventory files,
```
Comment (cp5555) [inline docs/getting-started/run-superbench.md:7]: What's the meaning of this sentence?

File: docs/getting-started/run-superbench.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+it will be easy to run SuperBench over all managed nodes.
+
+## Deploy
+
+We need to deploy SuperBench environments to all managed nodes first,
```
Comment (cp5555) [inline docs/getting-started/run-superbench.md:12]: environment without s?

File: docs/getting-started/run-superbench.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+it will access all nodes, pull container image and prepare container.
```
Comment (cp5555) [inline docs/getting-started/run-superbench.md:13]: Leveraging SuperBench `deploy` command, we can easily deploy SuperBench environment to all managed nodes. After running the command, SuperBench will automatically access all nodes, pull container image and prepare container.

File: docs/getting-started/run-superbench.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+If you are using password for SSH and cannot specify private key in inventory,
+or your private key requires a passphase before use, you can do
+```bash
+sb deploy -f remote.ini --host-password [password]
+```
+
+## Run
+
+After deployment, you can start to run the SuperBench benchmarks on all managed nodes.
```
Comment (cp5555) [inline docs/getting-started/run-superbench.md:29]: After deployment, you can start to run the SuperBench benchmarks on all managed nodes using SuperBench `run` command.

File: docs/getting-started/installation.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+SuperBench is used to run validations for AI infrastructure,
+thus you need to prepare one or multiple __managed nodes__ which are going to be validated,
+and one __control node__ which is used to run SuperBench commands.
+
+Usually __managed nodes__ are GPU nodes with high speed inter-connection, while __control node__ could be a CPU node.
+
+:::tip Tips
+It is fine if you have only one GPU node and want to try SuperBench on it.
+Control node could also be one of the managed nodes.
```
Comment (cp5555) [inline docs/getting-started/installation.md:15]: Control node and managed node can co-locate on the same machine.

File: docs/getting-started/installation.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+* Latest version of Linux, you're highly encouraged to use Ubuntu 18.04 or later.
+* Compatible GPU drivers should be install correctly.
+  * For NVIDIA GPUs, driver version can be checked by running `nvidia-smi`.
+* [Docker CE](https://docs.docker.com/engine/install/) version 19.03 or later (which can be checked by running `docker --version`).
+* GPU support in Docker.
+  * For NVIDIA GPUs, install
+  [nvidia-container-toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#setting-up-nvidia-container-toolkit).
+
+## Control node
```
Comment (cp5555) [inline docs/getting-started/installation.md:32]: Shall we move control node before managed node?

File: docs/cli.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+ |_____/ \__,_| .__/ \___|_|  |____/ \___|_| |_|\___|_| |_|
+              | |
+              |_|
+
+Welcome to the SB CLI!
+```
+
+## SuperBench CLI commands
+
+Below is the full list of SuperBench CLI commands and their usages:
```
Comment (cp5555) [inline docs/cli.md:25]: The following list shows `sb` command usage and examples

File: README.md (hunk 1 above)
Comment (cp5555) [inline README.md:19]: Only this two?

File: README.md [hunk 11]
```diff
@@ -9,304 +9,14 @@
 | cpu-unit-test | [![Build Status](https://dev.azure.com/msrasrg/SuperBenchmark/_apis/build/status/microsoft.superbenchmark?branchName=main)](https://dev.azure.com/msrasrg/SuperBenchmark/_build/latest?definitionId=77&branchName=main) |
 | gpu-unit-test | [![Build Status](https://dev.azure.com/msrasrg/SuperBenchmark/_apis/build/status/cuda-unit-test?branchName=main)](https://dev.azure.com/msrasrg/SuperBenchmark/_build/latest?definitionId=80&branchName=main) |
 
+__SuperBench__ is a validation and profiling tool for AI infrastructure.
 
-**SuperBench** is a validation and profiling tool for AI infrastructure, which supports:
+Check SuperBench website for more details.
```
Comment (cp5555) [inline README.md:14]: Do we plan to add link in next PR?

File: docs/cli.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+#### Global arguments
+
+| Name | Default | Description |
+| --- | --- | --- |
+| `--help` `-h` | N/A | Show help message. |
+
+#### Examples
+
+Execute GPT2 model benchmark only in default benchmarking configuration:
```
Comment (cp5555) [inline docs/cli.md:90]: Execute GPT2 model benchmark in default configuration:

File: docs/cli.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+| `--help` `-h` | N/A | Show help message. |
+
+#### Examples
+
+Deploy image `superbench/cuda:11.1` to all nodes in `./host.yaml`:
+```bash title="SB CLI"
+sb deploy --docker-image superbench/cuda:11.1 --host-file ./host.yaml
+```
+
+### `sb exec`
```
Comment (cp5555) [inline docs/cli.md:67]: Do we recommend to use `sb exec` for local run?

File: docs/getting-started/configuration.md [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+        - vgg11
+        - vgg13
+        - vgg16
+        - vgg19
+      parameters:
+        <<: *common_model_config
+        batch_size: 128
+```
+
+By default, all benchmarks specified in above config will be run if no custom config provided.
```
Comment (cp5555) [inline docs/getting-started/configuration.md:100]: By default, all benchmarks in default configuration will be run if you don't specify customized configuration.

File: docs/getting-started/configuration.md [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+          batch_size: 128
+  ```
+
+## Ansible Inventory
+
+SuperBench leverages [Ansible](https://docs.ansible.com/ansible/latest/) to run benchmarking workloads on managed nodes,
+you need to provide an [inventory](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html) file
+to configure host list for managed nodes.
+
+Here're some basic examples you can start from.
```
Comment (cp5555) [inline docs/getting-started/configuration.md:133]: Here're some basic examples as your starting point.

File: docs/getting-started/configuration.md [hunk 16]
```diff
@@ -0,0 +1,154 @@
+---
```
Comment (cp5555) [inline docs/getting-started/configuration.md:1]: After reading this document, I don't know whether this configuration is only for `sb run` or not. If yes, why do you recommend `sub execute`?

File: docs/cli.md (hunk 13 above)
Comment (abuccts) [reply docs/cli.md:67]: no

File: README.md (hunk 11 above)
Comment (abuccts) [reply README.md:14]: yes

File: README.md (hunk 1 above)
Comment (abuccts) [reply README.md:19]: no need to add all entries? just some important ones

File: docs/getting-started/run-superbench.md (hunk 4 above)
Comment (abuccts) [reply docs/getting-started/run-superbench.md:7]: [this one](https://ell.stackexchange.com/questions/57471/the-meaning-of-having-done-my-homework-i-will-go-home#:~:text=Having%20done%2FHaving%20finished%20is,carry%20out%20the%20second%20action.)?

File: docs/introduction.md (hunk 3 above)
Comment (abuccts) [reply docs/introduction.md:27]: it's not in [CommonMark](https://commonmark.org/), but one of the markdown extensions
e.g., syntax in https://docs.microsoft.com/en-us/contribute/markdown-reference

File: docs/cli.md (hunk 13 above)
Comment (abuccts) [reply docs/cli.md:67]: not yet

File: docs/getting-started/configuration.md (hunk 16 above)
Comment (abuccts) [reply docs/getting-started/configuration.md:1]: do not have scenarios for exec yet

File: README.md (hunk 1 above)
Comment (cp5555) [reply README.md:19]: Remove it directly.

**Review by cp5555 — APPROVED** [review]

File: docs/benchmarks/micro-benchmarks.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+  <tbody>
+    <tr align="center" valign="bottom">
+      <td>
+      </td>
+      <td>
+        <b>Micro Benchmark</b>
+        <img src={require('../assets/bar.png').default}/>
+      </td>
+      <td>
+        <b>Model Benchmark</b>
```
Comment (guoshzhao) [inline docs/benchmarks/micro-benchmarks.md:30]: why don't remove 'Model Benchmark'?

File: docs/benchmarks/model-benchmarks.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+
+## Benchmarking metrics
+
+<table>
+  <tbody>
+    <tr align="center" valign="bottom">
+      <td>
+      </td>
+      <td>
+        <b>Micro Benchmark</b>
```
Comment (guoshzhao) [inline docs/benchmarks/model-benchmarks.md:26]: why don't remove 'Micro Benchmark'?

File: README.md [hunk 19]
```diff
@@ -9,304 +9,9 @@
...
-#### Contribution steps
-
-If you would like to contribute to the project, please follow below steps of joint development on GitHub.
-
-1. `Fork` the repo first to your personal GitHub account.
-2. Checkout from main branch for feature development.
-3. When you finish the feature, please fetch the latest code from origin repo, merge to your branch and resolve conflict.
-4. Submit pull requests to origin main branch.
-5. Please note that there might be comments or questions from reviewers. It will need your help to update the pull request.
+_Check SuperBench website for more details._
```
Comment (guoshzhao) [inline README.md:14]: What does Superbench website mean? Do we need to add a link?

File: docs/benchmarks/micro-benchmarks.md [hunk 20]
```diff
@@ -0,0 +1,113 @@
+---
```
Comment (guoshzhao) [inline docs/benchmarks/micro-benchmarks.md:1]: The display of id is by designed? It's a little weird: [micro-benchmarks.md](https://github.com/microsoft/superbenchmark/blob/33414336d2a727111482c2ed8c99f593c033ffda/docs/benchmarks/micro-benchmarks.md)

File: README.md (hunk 19 above)
Comment (abuccts) [reply README.md:14]: link is not ready

File: docs/benchmarks/model-benchmarks.md (hunk 18 above)
Comment (abuccts) [reply docs/benchmarks/model-benchmarks.md:26]: it's placeholder, to be updated by tingting

File: docs/benchmarks/micro-benchmarks.md (hunk 17 above)
Comment (abuccts) [reply docs/benchmarks/micro-benchmarks.md:30]: it's placeholder, to be updated by tingting

File: docs/benchmarks/micro-benchmarks.md (hunk 20 above)
Comment (abuccts) [reply docs/benchmarks/micro-benchmarks.md:1]: it's markdown frontmatter, [for example](https://jekyllrb.com/docs/front-matter/)
this is doc source, not meant to be viewed by user directly, like [other microsoft docs](https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/service-fabric/service-fabric-cli.md)

Comment (guoshzhao) [reply docs/benchmarks/micro-benchmarks.md:1]: The term 'id' here is weird to me. Can we use 'title'?

**Review by guoshzhao — APPROVED** [review]

File: docs/benchmarks/micro-benchmarks.md (hunk 20 above)
Comment (abuccts) [reply docs/benchmarks/micro-benchmarks.md:1]: `id` is a field used when generating layout for website
there does exist a `title` field, by default it's the first header's content

**Pull request merged**